docs/map.xsd

```diff
@@ -247,22 +452,33 @@
 </xs:attributeGroup>
 
 <xs:attributeGroup name="tile.data.layer">
-  <xs:attribute name="gid" type="xs:nonNegativeInteger" use="required"/>
+  <xs:attribute name="gid" type="tileId" use="required"/>
 </xs:attributeGroup>
 
 <xs:attributeGroup name="objectgroup">
+  <!--Not sure if missing x,y which are defaulted to 0 and unchangable-->
   <xs:attribute name="name" type="nameT" use="required"/>
-  <xs:attribute name="width" type="xs:nonNegativeInteger" use="required"/>
-  <xs:attribute name="height" type="xs:nonNegativeInteger" use="required"/>
+  <xs:attribute name="width" type="pixelT" use="required"/>
+  <xs:attribute name="height" type="pixelT" use="required"/>
+  <xs:attribute name="color" type="hexColourT" use ="required"/>
+  <xs:attribute name="opacity" type="opacityT" use="optional" default="1"/>
+  <xs:attribute name="visible" type="xs:boolean" use="optional" default="1"/>
+  <xs:attribute name="offsetx" type="xs:integer" use="optional" default="0"/>
+  <xs:attribute name="offsety" type="xs:integer" use="optional" default="0"/>
+  <xs:attribute name="draworder" type="draworderT" use="optional" default="topdown"/>
 </xs:attributeGroup>
 
 <xs:attributeGroup name="object">
+  <xs:attribute name="id" type="xs:nonNegativeInteger" use="required"/>
```

Comment:
Ok. Though it seems weird to have an id (which is usually a unique key for some data) to be optional. But I'll mark it as optional :)